### ISSUE TITLE
cannot install rust-src component for nightly-2020-07-21

### ISSUE THREAD

**Issue by ehuss** [rust-lang/rust#74577]
Not sure if this is due to the packaging, or if it is a rustup bug, but I cannot install the rust-src component for the latest nightly-2020-07-21.

```
> rustup toolchain install nightly-2020-07-21 --profile=minimal -c rust-src
info: syncing channel updates for 'nightly-2020-07-21-x86_64-unknown-linux-gnu'
info: latest update on 2020-07-21, rust version 1.47.0-nightly (f9a308636 2020-07-20)
info: downloading component 'cargo'
  5.2 MiB /   5.2 MiB (100 %)   1.7 MiB/s in  3s ETA:  0s
info: downloading component 'rust-src'
  2.1 MiB /   2.1 MiB (100 %)   1.7 MiB/s in  1s ETA:  0s
info: downloading component 'rust-std'
 19.7 MiB /  19.7 MiB (100 %)   1.5 MiB/s in 13s ETA:  0s
info: downloading component 'rustc'
 51.0 MiB /  51.0 MiB (100 %) 896.0 KiB/s in 59s ETA:  0s
info: installing component 'cargo'
info: Defaulting to 500.0 MiB unpack ram
info: installing component 'rust-src'
info: rolling back changes
error: tar entry kind 'Symlink' is not supported
```

Fails on macOS and Linux with the same error.

I'll try to dig in some tonight, but figured I'd file this to see if anyone has any ideas.  If I had to take a wild guess, I would say it may be related to #74520.

`rustup 1.22.1 (b01adbbc3 2020-07-08)`

**Comment by lzutao** [comment]
Rustup doesn't support symlink entry in tar file: https://github.com/rust-lang/rustup/blob/ac1a8c415b5c2886275ca92a3a8f3a4cd0587711/src/dist/component/package.rs#L397-L407
There are two symlinks in backtrace:
```
% fd --type l . src/backtrace 
src/backtrace/crates/backtrace-sys/LICENSE-APACHE
src/backtrace/crates/backtrace-sys/LICENSE-MIT
```

**Comment by Mark-Simulacrum** [comment]
cc @alexcrichton

**Comment by Mark-Simulacrum** [comment]
Once this is fixed we should look into checking that we're not creating these tarballs on the dist builders (rather than failing "late" when rustup fails to install)

**Comment by matthiaskrgr** [comment]
Getting the same with `rust-toolchain-installer-master`:
`error: unsupported tar entry: Symlink`

**Comment by ehuss** [comment]
Posted a fix at #74578.
